src/backend/expungeservice/endpoints/search.py

```diff
@@ -0,0 +1,58 @@
+from flask.views import MethodView
+from flask import request, current_app
+
+from expungeservice.request import check_data_fields
+from expungeservice.request.error import error
+from expungeservice.crawler.crawler import Crawler
+from expungeservice.expunger.expunger import Expunger
+from expungeservice.endpoints.auth import user_auth_required
+from expungeservice.serializer import ExpungeModelEncoder
+from expungeservice.crypto import DataCipher
+
+
+class Search(MethodView):
+
+    @user_auth_required
+    def post(self):
+        request_data = request.get_json()
+
+        if request_data is None:
+            error(400, "No json data in request body")
+
+        check_data_fields(request_data, ["first_name", "last_name",
+                                 "middle_name", "birth_date"])
+
+        cipher = DataCipher(
+            key=current_app.config.get("JWT_SECRET_KEY"))
+
+        decrypted_credentials = cipher.decrypt(request.cookies["oeci_token"])
+
+        crawler = Crawler()
+
+        login_result = crawler.login(
+            decrypted_credentials["oeci_username"],
+            decrypted_credentials["oeci_password"],
+            close_session=False)
+
+        record = crawler.search(
+            request_data["first_name"],
+            request_data["last_name"],
+            request_data["middle_name"],
+            request_data["birth_date"])
+
+        if record.cases:
```

Comment:
There is no need to check if the cases array is empty. The expunger doesn't break if it is passed a record object with an empty array of cases.  I'd recommend to remove this line.